examples/cfd/aero_graph_net/train.py

```diff
@@ -0,0 +1,262 @@
+        self.optimizer.zero_grad()
+        losses = self.forward(batch)
+        self.backward(losses["total"])
+        self.scheduler.step()
+        return losses
+
+    def forward(self, batch):
+        # forward pass
+        graph = batch["graph"]
+        with self.autocast():
+            pred = batch_as_dict(self.model(graph.ndata["x"], graph.edata["x"], graph))
+            # Graph data (e.g. p and WSS) loss.
+            graph_loss = self.loss.graph(pred["graph"], graph.ndata["y"])
+            losses = {"graph": graph_loss}
+            # Compute C_d loss, if requested.
+            if (pred_c_d := pred.get("c_d")) is not None:
+                c_d_loss = self.loss.c_d(pred_c_d, batch["c_d"])
+                losses["c_d"] = c_d_loss
+            # Get total loss and detach intermediate losses.
+            total_loss = sum(losses.values())
+            losses = {k: v.detach() for k, v in losses.items()}
+            losses["total"] = total_loss
+
+            return losses
+
+    def backward(self, loss):
+        # backward pass.
+        # If AMP is disabled, the scaler will fall back to the default behavior.
+        self.scaler.scale(loss).backward()
+        self.scaler.step(self.optimizer)
+        self.scaler.update()
+
+    @torch.no_grad()
+    def validation(self, epoch: int):
+        losses_agg = defaultdict(float)
+        for batch in self.validation_dataloader:
+            batch = batch_as_dict(batch, self.dist.device)
+            graph = batch["graph"]
+            pred = batch_as_dict(self.model(graph.ndata["x"], graph.edata["x"], graph))
+            pred_g, gt_g = self.dataset.denormalize(
+                pred["graph"], graph.ndata["y"], self.dist.device
+            )
+            losses_agg["graph"] += self.loss.graph(pred_g, gt_g)
+            if (pred_c_d := pred.get("c_d")) is not None:
+                losses_agg["c_d"] += self.loss.c_d(pred_c_d, batch["c_d"])
+
+        losses_agg["total"] = sum(losses_agg.values())
+
+        # Visualize last batch.
+        for vis in self.visualizers.values():
+            vis(graph, pred_g, gt_g, epoch, elogger)
+
+        # Log losses.
+        num_batches = len(self.validation_dataloader)
+        loss_str = []
+        for k, v in losses_agg.items():
+            loss = v / num_batches
+            elogger.log_scalar(f"val/loss/{k}", loss, epoch)
+            loss_str.append(f"{k}: {loss:6.4f}")
+
+        logger.info(f"Validation loss: {', '.join(loss_str)}")
+
+
+@hydra.main(version_base="1.3", config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+    # initialize distributed manager
+    DistributedManager.initialize()
+    dist = DistributedManager()
+
+    init_python_logging(cfg)
+    logger.info(f"Config summary:\n{OmegaConf.to_yaml(cfg, sort_keys=True)}")
+
+    torch.set_float32_matmul_precision("high")
+    torch.backends.cudnn.benchmark = True
+    torch.backends.cudnn.allow_tf32 = True
```

Comment:
I'm curious if you have made any accuracy comparisons between the models trained with and without TF32?

Reply:
No, not yet, but it would an interesting experiment.